admin-guide/en/appendix-1.-create-a-protected-ojs-3-sandbox-staged-with-git.md

```diff
@@ -23,19 +23,16 @@ The README here: [https://github.com/pkp/ojs](https://github.com/pkp/ojs) has in
    ```
    cd <httpd-docs-folder>
    BRANCH='stable-3_2_1'
-   git clone -n https://github.com/pkp/ojs.git ./
-   git checkout -b $BRANCH --no-track origin/$BRANCH
-   cp config.TEMPLATE.inc.php config.inc.php
+   git clone -b $BRANCH --single-branch https://github.com/pkp/ojs.git ./
+   git branch --unset-upstream $BRANCH
    chmod -R 755 *
-   chmod 600 config.inc.php
    ```
 
 3. Fetch corresponding PKP library and checkout stable branch from GitHub, making sure that the branch corresponds to the same branch referenced above:
 
    ```
    git submodule update --init --recursive
-   cd lib/pkp
-   git checkout -b $BRANCH --no-track origin/$BRANCH
+   git submodule foreach --recursive git checkout -b $BRANCH
```

Comment:
I don't think we're guaranteed to have `$BRANCH` defined in all submodules (especially plugins).  On the other hand, any checkouts after the initial `git submodule update` seem unnecessary.

Can you share more about what situation this addresses or fixes?

Reply:
It doesn't checkout the name that has those submodules as such, but the commits that the branch of OJS is pointing at. So it's just a name for the detached commit that is pointed out. I just simplified what was already written.

But I do agree with you that the step of checking out submodules seems unnecessary. If this is the case, it could even be removed the steps 1 or 2 steps following updating the submodules.

Reply:
I read `git submodule foreach --recursive git checkout -b $BRANCH` as saying:
for each submodule in repo
  checkout the submodule at branch `$BRANCH`
     recursively

The original code, without the foreach implications (`cd lib/pkp; git checkout -b $BRANCH --no-track origin/$BRANCH`) would only checkout the lib/pkp submodule at `$BRANCH`.

Since it seems both are unnecessary, can you push a commit which removes the lines in favor of just leaving `git submodule update --init --recursive`?

Reply:
> I read `git submodule foreach --recursive git checkout -b $BRANCH` as saying:
> 
>     * for each submodule in repo
>       
>       * checkout the submodule at branch `$BRANCH`
>         
>         * recursively
> 
> 
> The original code, without the foreach implications (`cd lib/pkp; git checkout -b $BRANCH --no-track origin/$BRANCH`), would only checkout the lib/pkp submodule at `$BRANCH`.
> 
> Since it seems both are unnecessary, can you push a commit which removes the lines in favor of just leaving `git submodule update --init --recursive`?

Done